packages/core/src/session/chat-session.ts

```diff
@@ -134,17 +151,21 @@ export class ChatSession {
             sessionManager: import('./session-manager.js').SessionManager;
         },
         public readonly id: string,
-        logger: IDextoLogger
+        logger: IDextoLogger,
+        agentConfig?: import('../agent/schemas.js').AgentConfig
```

Comment:
we actually dont need an agentconfig argument

constructor already accepts custom `services` argument

override those when creating the chatsession instead

this way sessionmanager handles the service override and we don't need any extra management logic in chatsession